### DISCUSSION TITLE
Is there any way to download files in parallel from different nodes, just like bittorent

### DISCUSSION THREAD

**Discussion by silicology1** [libp2p/rust-libp2p#5508] [Q&A]
The following file sharing example, only connects to the [single peer to fetch the complete file](https://github.com/libp2p/rust-libp2p/blob/master/examples/file-sharing/src/main.rs#L102). What if I connect to multiple peers, and fetch part of file from different peers, just like bittorrent does.

**Comment by guillaumemichel** [comment]
It is certainly possible to implement it on top of libp2p, but I don't believe that such an example exists yet. Feel free to implement it and add it to the existing [examples](https://github.com/libp2p/rust-libp2p/tree/master/examples).
